### PR TITLE
Integrate AdvantageKit into 2023 Codebase

### PR DESCRIPTION
**Pull request by JohnGilb**
- Apex Subsystems are added to the `dataFrameRefreshables` in `Robot.java`
- Many ephemeral properties converted to logged objects
- Fix for robot going off-target when both translating and rotating. Works in simulator, has some minor effect on real robot.

### CODE REVIEW

File: src/main/java/competition/subsystems/drive/swerve/SwerveSteeringSubsystem.java [hunk 1]
```diff
@@ -342,5 +342,17 @@ public void periodic() {
         }
 
         currentModuleHeading.set(getCurrentValue());
+
+        motorController.periodic();
```
Comment (JohnGilb) [inline src/main/java/competition/subsystems/drive/swerve/SwerveSteeringSubsystem.java:346]: I don't know how, but:
- This is the only significant difference between steering and drive parts of the module
- Only the steering fails to actuate
- The `periodic()` call does set PID values

All of this makes me suspicious.

Comment (JohnGilb) [reply src/main/java/competition/subsystems/drive/swerve/SwerveSteeringSubsystem.java:346]: Yeah, this is it. When the Spark is created it has its own internal set of PID values, which aren't updated by calls to SetP, SetD, etc. When Periodic is called, on the first call, it sets all PID values from its internal set.

File: src/main/java/competition/subsystems/drive/DriveSubsystem.java [hunk 2]
```diff
@@ -3,34 +3,40 @@
...
+import xbot.common.math.XYPair;
 import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.PropertyFactory;
-import xbot.common.properties.StringProperty;
 import xbot.common.properties.XPropertyManager;
 import xbot.common.subsystems.drive.BaseDriveSubsystem;
 import xbot.common.subsystems.pose.BasePoseSubsystem;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
```
Comment (stephenjust) [inline src/main/java/competition/subsystems/drive/DriveSubsystem.java:35]: Duplicate imports?

File: src/main/java/competition/subsystems/drive/swerve/SwerveDriveSubsystem.java [hunk 3]
```diff
@@ -44,43 +42,26 @@ public SwerveDriveSubsystem(SwerveInstance swerveInstance, XCANSparkMaxFactory s
...
-                // for description of the different status frames. kStatus2 is the only frame with data needed for software PID.
-
-                this.motorController.getInternalSparkMax().setPeriodicFramePeriod(PeriodicFrame.kStatus0, 500 /* default 10 */);
-                this.motorController.getInternalSparkMax().setPeriodicFramePeriod(PeriodicFrame.kStatus1, 20 /* default 20 */);
-                this.motorController.getInternalSparkMax().setPeriodicFramePeriod(PeriodicFrame.kStatus2, 20 /* default 20 */);
-                this.motorController.getInternalSparkMax().setPeriodicFramePeriod(PeriodicFrame.kStatus3, 500 /* default 50 */);
-                
-                this.motorController.clearFaults();
-            }
+            this.motorController.setupStatusFramesIfControllerHasRecentRecently(500, 20, 20, 500);
```
Comment (stephenjust) [inline src/main/java/competition/subsystems/drive/swerve/SwerveDriveSubsystem.java:64]: Awkward method name

**Review by stephenjust — APPROVED** [review]
Minor feedback, otherwise looks good

**Pull request merged**